src/app/modules/shared_models/message_model.nim

```diff
@@ -244,12 +244,12 @@ QtObject:
       if(self.items[i].responseToMessageWithId == messageId):
         result.add(self.items[i].id)
 
-  proc findIndexBasedOnTimestampToInsertTo(self: Model, timestamp: int64, localTimestamp: int64): int =
+  proc findIndexBasedOnClockToInsertTo(self: Model, clock: int64, id: string): int =
     for i in 0 ..< self.items.len:
-      if timestamp > self.items[i].timestamp:
+      if clock > self.items[i].clock:
         return i
-      elif timestamp == self.items[i].timestamp:
-        if localTimestamp > self.items[i].localTimestamp:
+      elif clock == self.items[i].clock: # break ties by message id
+        if id > self.items[i].id:
```

Comment:
@osmaczko we should not compare strings like this? I guess that you should convert them to number first?

Reply:
that's consistent with what we do in mobile/status-go, if the clock is different, they are compared as strings, also I think comparing them as integers would give you the exact same, since they have equal length, so you are always comparing the most relevant digit